### ISSUE TITLE
Proposal: Add "consistent" option to array-element-newline

### ISSUE THREAD

**Issue by robertrossmann** [eslint/eslint#9457]
**Tell us about your environment**

* **ESLint Version:** 4.9.0
* **Node Version:** 8.7.0
* **npm Version:** 5.4.2

I would like to propose to add a third, new option, `consistent`, to the rule [array-element-newline](https://eslint.org/docs/rules/array-element-newline).

### Expected behaviour

Given `'array-element-newline': ['warn', 'consistent']`, a newline would be required between all array elements if any of the elements are placed on a separate line.

#### Good

```js
// 'array-element-newline': ['warn', 'consistent']
const arr = [1,2,3]
const arr = [
  1, 2, 3
]
const arr = [
  1,
  2,
  3,
]
```

#### Bad

```js
// 'array-element-newline': ['warn', 'consistent']
const arr = [
  1, 2,
  3,
]
```

#### Autofix

If implemented, the autofix would separate all elements onto a separate line if any of the elements are on a separate line. The autofix would never put all elements into a single line.

### Current behaviour

The rule currently only supports only:

- no newlines between elements
- `num` elements-based decision about elements placement
- all elements on their own line

-----

This follows the idea recently implemented in #9206 for `array-bracket-newline` and is tangentially related to #9054.

**Comment by robertrossmann** [comment]
Thanks @platinumazure! I wonder if I can start working on a PR or we need more votes here to accept this...? Is there anything I can do to further this?

**Comment by ilyavolodin** [comment]
@robertrossmann Every proposal requires 3 members of the team to support it, and one more member to champion it. I'll support this proposal. @eslint/eslint-team anyone else wants to support this?

**Comment by Aladdin-ADD** [comment]
I like this option! 👍

**Comment by platinumazure** [comment]
I'll champion. Just needs one more :+1: from the @eslint/eslint-team!

**Comment by platinumazure** [comment]
@eslint/eslint-team Can we get one more :+1: for this proposal? If not, I'll have to close. Thanks!

**Comment by kaicataldo** [comment]
Apologies for the delay. This is now accepted.
